app/transactions/members/update_transaction.rb

```diff
@@ -3,39 +3,56 @@
 class Members::UpdateTransaction
   include Dry::Transaction
 
+  tee :params
   tee :extract_ids
   step :assign_static_slot
+  step :save_in_history
   step :update_member
 
   private
 
-  def extract_ids(input)
-    return Success(input) if input[:permitted_params][:static_slots_attributes].nil?
+  def params(input)
+    @current_member = input[:current_member]
+    @static_slots_attributes = input[:permitted_params][:static_slots_attributes]
+    input[:permitted_params].delete('static_slots_attributes')
+    @member_params = input[:permitted_params]
+  end
+
+  def extract_ids
+    return Success if @static_slots_attributes.nil?
 
-    static_slot_ids = []
-    input[:permitted_params][:static_slots_attributes].values.each do |value|
-      static_slot_ids << value['id']
+    @static_slot_ids = []
+    @static_slots_attributes.each_pair do |_key, value|
+      @static_slot_ids << value['id']
     end
-    input.merge!({ static_slot_ids: static_slot_ids })
-    Success(input)
+    @static_slot_ids.uniq
```

Comment:
what is the point of this?
If you want to mutate the `@static_slot_ids` value, you should do `.uniq!`